api/client/registry.go

```diff
@@ -91,6 +91,10 @@ func (cli *DockerCli) ConfigureAuth(flUser, flPassword, serverAddress string, is
 		cli.in = os.Stdin
 	}
 
+	if !isDefaultRegistry {
```

Comment:
@aaronlehmann BTW, `RegistryAuthenticationPrivilegedFunc` hardcodes false just because it needs not to pop up:
> Login with your Docker ID to push and pull images from Docker Hub. If you don'    t have a Docker ID, head over to https://hub.docker.com to create one.
When prompting interactively... which seems wrong so fixing..